app/locales/locales.ts

```diff
@@ -31,19 +33,12 @@ export const locales = ["de", "fr", "it", "en"] as const;
 
 export type Locale = typeof locales[number];
 
-i18n.loadLocaleData({
-  de: { plurals: pluralsDe },
-  fr: { plurals: pluralsFr },
-  it: { plurals: pluralsIt },
-  en: { plurals: pluralsEn },
-});
-i18n.load({
-  de: catalogDe,
-  fr: catalogFr,
-  it: catalogIt,
-  en: catalogEn,
-});
-i18n.activate(defaultLocale);
+export const activate = async (locale: typeof locales[number]) => {
+  i18n.loadLocaleData(locale, localeData[locale]);
+  const messages = await import(`./${locale}/messages`);
+  i18n.load(locale, messages);
+  i18n.activate(defaultLocale);
+};
```

Comment:
Wouldn't this mean that the messages will now be imported each time when a user switches a locale?

Reply:
I removed this, it was part of an experimentation to load locales asynchronously, but it shouldn't be in this PR. 

---

To answer your original question, I would think that the asynchronous `import` gets converted to a call that fetches from webpack's runtime cache, and if the module has not been loaded, it would load it across the network. I will check this :)